includes/tuque_wrapper.inc

```diff
@@ -239,8 +239,16 @@ class IslandoraFedoraObject extends FedoraObject {
       islandora_invoke_datastream_hooks(ISLANDORA_DATASTREAM_INGESTED_HOOK, $object->models, $datastream->id, $object, $datastream);
       return $ret;
     }
+    catch (DatastreamExistsException $e) {
+      $message = islandora_deprecated('7.x-1.10', t('The datastream @dsid already exists on the object: @pid. After the 7.x-1.10 release this will throw an exception.', array(
+        '@dsid' => $datastream->id,
+        '@pid' => $object->id,
+      )));
+      trigger_error(filter_xss($message), E_USER_DEPRECATED);
+      return FALSE;
```

Comment:
I feel like this could be handled somewhat more gracefully... will also be less-than ideal to write code making use of the exceptions... though not sure how it would be controlled... a Drupal variable?

... Almost wanting to avoid the separate `catch` entirely, just push down into the general exception bit below in an `if` block... Something like:
```php
[...]
catch (Exception $e) {
  if ($e instanceof DatastreamExistsException) {
    // Message; and,
    return FALSE;
  }
  // Do the present watchdog and rethrow.
}
[...]
```
if we wanted to introduce a variable to control whether it returns `FALSE` or throws, could then do something like:
```php
[...]
catch (Exception $e) {
  if ($e instanceof DatastreamExistsException && variable_get('islandora_deprecation_return_false_when_datastream_exists', TRUE)) {
    // Message; and,
    return FALSE;
  }
  // Do the present watchdog and rethrow.
}
[...]
```
Not sure where/how we'd want to allow it to be set to `FALSE`...

... either way, having it as one little condition to wipe may make it easier to strip out when the times comes.